http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -301,31 +306,70 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
   private def createDisk(params: CreateAzureRuntimeParams, leoAuth: Authorization)(implicit
     ev: Ask[F, AppContext]
   ): F[CreateDiskResponse] =
-    for {
-      ctx <- ev.ask
-      diskOpt <- persistentDiskQuery.getById(params.runtimeConfig.persistentDiskId).transaction
-      disk <- F.fromOption(diskOpt, new RuntimeException("no disk found"))
-      common = getCommonFields(ControlledResourceName(disk.name.value),
-                               config.runtimeDefaults.diskControlledResourceDesc,
-                               params.runtime.auditInfo.creator,
-                               None
-      )
-      request: CreateDiskRequest = CreateDiskRequest(
-        params.workspaceId,
-        common,
-        CreateDiskRequestData(
-          // TODO: AzureDiskName should go away once DiskName is no longer coupled to google2 disk service
-          AzureDiskName(disk.name.value),
-          disk.size,
-          params.runtimeConfig.region
-        )
-      )
-      diskResp <- wsmDao.createDisk(request, leoAuth)
-      _ <- controlledResourceQuery
-        .save(params.runtime.id, diskResp.resourceId, WsmResourceType.AzureDisk)
-        .transaction
-      _ <- persistentDiskQuery.updateStatus(disk.id, DiskStatus.Ready, ctx.now).transaction
-    } yield diskResp
+    params.useExistingDisk match {
+      case true =>
+        for {
+          ctx <- ev.ask
+          diskOpt <- persistentDiskQuery.getById(params.runtimeConfig.persistentDiskId).transaction
+          disk <- F.fromOption(
+            diskOpt,
+            new RuntimeException(s"Disk id:${params.runtimeConfig.persistentDiskId} not found")
+          )
+          resourceId <- F.fromOption(
+            disk.wsmResourceId,
+            AzureRuntimeCreationError(
+              params.runtime.id,
+              params.workspaceId,
+              s"No associated resourceId found for Disk id:${params.runtimeConfig.persistentDiskId.value}",
+              params.useExistingDisk
+            )
+          )
+          diskResourceOpt <- controlledResourceQuery
+            .getWsmRecordFromResourceId(resourceId, WsmResourceType.AzureDisk)
+            .transaction
+          wsmDisk <- F.fromOption(
+            diskResourceOpt,
+            AzureRuntimeCreationError(
+              params.runtime.id,
+              params.workspaceId,
+              s"WSMResource:${resourceId} not found for disk id:${params.runtimeConfig.persistentDiskId.value}",
+              params.useExistingDisk
+            )
+          )
+          _ <- persistentDiskQuery.updateStatus(disk.id, DiskStatus.Ready, ctx.now).transaction
```

Comment:
Hmm... why are we updating it to `Ready` here? Don't we check in front leo if its `Ready`, and if not, throw an error? I would expect that behavior to occur here as well...

Reply:
good catch, this was leftover from when I wasn't checking the status in front leo, I removed it